doc: thingy53: nrf52: Update docs after config simplification (USB CDC ACM, FOTA) (#10256)
nrfconnect/sdk-nrf: doc/nrf/working_with_nrf/nrf52/developing.rst

FOTA over Bluetooth Low Energy
==============================

FOTA updates are supported using MCUmanager's Simple Management Protocol (SMP) over Bluetooth.
The application acts as a GATT server and allows the connected Bluetooth Central to perform the firmware update.

> Sample must support Bluetooth peripheral role (:kconfig:option:`CONFIG_BT_PERIPHERAL`) in order to use this feature.

Author: It refers to all samples so applied your change

The application supports SMP handlers related to:

* Image management.
* Operating System (OS) management -- used to reboot the device after the firmware upload is complete.

> * Erasing settings partition -- used to ensure that a new application is not booted with incompatible content in the settings partition written by the previous application.

Reviewer: If possible, I guess this section can be shared between the Thingy:53 ug and nRF52/53 DK UGs

Let's wait with this for the technical writers input

Author: I think so too. The only problem is the `CONFIG_BOOTLOADER_MCUBOOT` which is needed to be manually in nRF52/53 DKs and is already enabled by default fot Thingy:53

Reviewer 2: So, if there is some part that is good for all of the listed docs (whether it is a whole section or just a subset of it), you can mark such area with the `.. fota_something_start` in the beginning and `.. fota_something_end` at the end of the area. Then just use that marked area in other docs the way you had in your earlier comment above.

To enable support for FOTA updates, do the following:

* Use MCUboot as the upgradable bootloader (:kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT` must be enabled).
  For more information, go to the :doc:`mcuboot:index-ncs` page.

> * Enable :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` Kconfig option, which implies configuration of:

Reviewer: Also here

(Let's wait with this improvement suggestion for the technical writer's input)

> * Enable :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` Kconfig option, which implies configuration of:

Reviewer 2: ```suggestion
* Enable the :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` Kconfig option, which implies configuration of the following:
```

   * All of the SMP command handlers mentioned in the previous paragraph.
   * SMP BT reassembly feature.

>    * :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU_SPEEDUP` option -- automatically extends the Bluetooth buffers, which allows to speed up the FOTA transfer over Bluetooth, but also increases RAM usage.

Reviewer 2: ```suggestion
   * The :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU_SPEEDUP` Kconfig option -- automatically extends the Bluetooth buffers, which allows to speed up the FOTA transfer over Bluetooth, but also increases RAM usage.
```

>    * :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU_SPEEDUP` option -- automatically extends the Bluetooth buffers, which allows to speed up the FOTA transfer over Bluetooth, but can also increase RAM usage.

Reviewer 3: `also increases` (it always increases the RAM usage actually :) )

>    * :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU_SPEEDUP` option -- automatically extends the Bluetooth buffers, which allows to speed up the FOTA transfer over Bluetooth, but can also increase RAM usage.

Reviewer 3: I would also inform that Bluetooth buffers configuration is also automatically applied to the network core child image (maybe you could mention which network core images are supported aswell - hci_rpmsg and multiprotocol_rpmsg).

Author: I think that we should with that for technical writer's input like @kapi-no suggested in https://github.com/nrfconnect/sdk-nrf/pull/10256#discussion_r1105847614

> If necessary, any of the implied options or defaulted values by :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` can be modified.

Reviewer 3: `... implied options or defaulted values introduced by :kconfig:option: ...`

> If necessary, any of the implied options or defaulted values introduced by :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` can be modified.

Reviewer 2: ```suggestion
If necessary, you can modify any of the implied options or defaulted values introduced by the :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` Kconfig option.
```

> You can either add the above Kconfig options to your :file:`prj.conf` file or inline them in a project build command.

Reviewer 3: to your application configuration files (there can be multiple application configuration files in Zephyr, configuration files for other build types can have different names too)

> You can either add the above Kconfig options to your application configuration files or inline them in a project build command.

Reviewer 2: ```suggestion
You can either add these Kconfig options to the configuration files of your application or have them inline in a project build command.
```

> See how it can be done for ref:`peripheral_lbs` sample:

Reviewer 2: ```suggestion
Here is an example of how you can build for the :ref:`peripheral_lbs` sample:
```

.. parsed-literal::
   :class: highlight

    west build -b *build_target* -- -DCONFIG_BOOTLOADER_MCUBOOT=y -DCONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU=y

> Once you have completed these steps, when you connect to the device, you will see the SMP Service enabled with the ``UUID 8D53DC1D-1DB7-4CD3-868B-8A527460AA84``.

Reviewer 2: I don't see any steps.
Maybe something like:
When you connect to the device after the build has completed, the SMP Service is enabled with ....

Author: I added something like `... and the firmware has been programmed to it ...` to your proposal as it won't work without it, please check current version

> If you want to improve your application by adding SMP Service to advertising data, take a look at how it is done in :ref:`zephyr:smp_svr_sample`.

Reviewer 2: ```suggestion
If you want to add SMP Service to advertising data, refer to the :ref:`zephyr:smp_svr_sample`.
```

* Use MCUboot as the upgradable bootloader (:kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT` must be enabled).
  For more information, go to the :doc:`mcuboot:index-ncs` page.
* Enable :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` Kconfig option.

   * By default, :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU_SPEEDUP` option is implied when FOTA is enabled using mentioned option.
     This option automatically extends the Bluetooth buffers, which allows to speed up the FOTA transfer over Bluetooth, but can also increase RAM usage.
     You can disable this option manually if necessary.

> 

Reviewer: It would be great to add a section on how to build for a specific project like the Peripheral LBS sample. Something along these lines: 

_You can enable the FOTA feature in a single command (...) Please see how it can be done for the Peripheral LBS sample (replace the `<target>` with your target board):_

```
west build -b <target> -- -DCONFIG_BOOTLOADER_MCUBOOT=y -DCONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU=y
```
